pyccel/codegen/printing/fcode.py

```diff
@@ -307,15 +307,40 @@ def print_kind(self, expr):
         return constant_name
 
     def _handle_inline_func_call(self, expr, provided_args, assign_lhs = None):
-        """ Print a function call to an inline function
         """
+        Print a function call to an inline function.
+
+        Use the arguments passed to an inline function to print
+        its body with the passed arguments in place of the function
+        arguments.
+
+        Parameters
+        ----------
+        expr : FunctionCall
+            The function call which should be printed inline.
+
+        provided_args : List
+            A list of FunctionCallArguments.
+
+        assign_lhs : List
+            A list of lhs provided.
+
+        Returns
+        -------
+        str
+            The code for the inline function.
+        """
+
         scope = self.scope
         func = expr.funcdef
 
         # Print any arguments using the same inline function
         # As the function definition is modified directly this function
         # cannot be called recursively with the same FunctionDef
         args = []
+        if isinstance(expr, DottedFunctionCall):
+            if self.scope.find(expr.funcdef.cls_name, 'classes'):
+                args.append(expr.args[0])
```

Comment:
Do you know why `self` is not in the args? Can this be fixed in the call to `_handle_inline_func_call`?

Reply:
the `self` is removed here : https://github.com/pyccel/pyccel/blob/a332075ee1f8b7d3bd0f4390a01bb7af7343ac69/pyccel/codegen/printing/fcode.py#L2907-L2908

Reply:
Maybe we should use `expr.args` in the arguments to `_handle_inline_func_call` instead of removing `self` then using costly functions to rediscover it

Reply:
Fixed https://github.com/pyccel/pyccel/pull/1576/commits/c163e184f5d393e8fb5289b35159e4a8c413dced.